src/flinkStatementResultsManager.test.ts

```diff
@@ -64,8 +52,16 @@ describe("FlinkStatementResultsManager", () => {
       DEFAULT_RESULTS_LIMIT,
     );
 
-    // Wait for results to be processed
-    await new Promise((resolve) => setTimeout(resolve, 20));
+    // Wait for results to be processed, it should eventually become 10
+    while (manager.handleMessage("GetResultsCount", {}).total !== 10) {
+      await new Promise((resolve) => setTimeout(resolve, 20));
+    }
```

Comment:
We may want to revisit this test to add more stubbing along with https://sinonjs.org/releases/latest/fake-timers/ since real-waiting can get finnicky (especially in CI)